### PR TITLE
plat-2744 use tags for deployment

### PR DESCRIPTION
**Pull request by tfhartmann**
This PR  is the initial offering of using our "deploy from tags" module using hermes:terraform-x.x.x  there are a couple other changes in this spike that mostly were just done to bring it up to spec before setting up the circle example.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -14,9 +13,17 @@ if has vault; then
       TOKEN_RENEW=$(vault token-renew -format=json 2>/dev/null)
     else
       # authenticate
-      vault auth -method=github
+      vault login -method=github
```
Comment (hakamadare) [inline .envrc:16]: goodbye deprecation warning!

File: .envrc [hunk 2]
```diff
@@ -14,9 +13,17 @@ if has vault; then
...
     fi
   fi
 fi
 
+function get_vault_kv {
+  vault_path=$1
```
Comment (hakamadare) [inline .envrc:22]: for future enhancement: this function should probably check to make sure that there are two non-empty arguments passed in, and blow up if no vault_path is passed (andmaybealsonotbewritteninbash)

File: env-production/terraform.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 0.11.7"
```
Comment (hakamadare) [inline env-production/terraform.tf:2]: 0.11.8?

File: env-staging/terraform.tf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 0.11.7"
```
Comment (hakamadare) [inline env-staging/terraform.tf:2]: 0.11.8?

**Review by hakamadare — APPROVED** [review]
AWESOME

this is how i imagined our CircleCI configs looking.  i have some minor feedback but nothing that should block a merge to master.

File: env-production/terraform.tf (hunk 3 above)
Comment (tfhartmann) [reply env-production/terraform.tf:2]: Yeah I was trying not to guild the lily to much,  I was trying to stay focused on the task at hand, which was the deployment patter, rather then to much refactoring.... ug.. why is computer-ing so hard 😆

**Pull request merged**